internal/prober/http/http.go

```diff
@@ -252,6 +260,25 @@ func convertOAuth2Config(ctx context.Context, cfg *sm.OAuth2Config, logger zerol
 	return r, nil
 }
 
+// Overrides any user-provided headers with our own augmented values
+// for 'reserved' headers.
+func augmentHttpHeaders(check *sm.Check, additionalHeaders []string) {
+	headers := []string{}
+	for _, header := range check.Settings.Http.Headers {
+		name, _ := strToHeaderNameValue(header)
+
+		if strings.ToLower(name) == checkProbeIdHeader {
+			continue // users can't override this header with their own value.
+		}
+
+		headers = append(headers, header)
+	}
+	headers = append(headers, additionalHeaders...)
+
+	httpHeaders := &check.Settings.Http.Headers
+	*httpHeaders = headers
```

Comment:
I believe this should be equivalent to:
```suggestion
	check.Settings.Http.Headers = headers
```
As Golang [automatically](https://go.dev/tour/moretypes/4) dereferences it.